### PR TITLE
Dolfyn updates

### PR DESCRIPTION
**Pull request by jmcvey3**
A number of updates:
1. Fixes for TRDI reader: #182, #206, #207
2. Fix for VMDAS files with interleaved pings
3. Motion correction for duty-cycled ADVs
4. lncorporated logging module for into reader debuggers
5. Ensure datatypes are saved as float32 instead of float64 to reduce filesize
6. Added another ADCP clean function
7. Updated docstrings to mhkit formatting
8. Other minor bugfixes

### CODE REVIEW

**Comment by ssolson** [comment]
James thank you for submitting the PR. To start the review I have a couple of questions about large log files:

1. 1.16 MB `examples/data/dolfyn/test_data/AWAC_test01.log`
2. 6.74 MB `examples/data/dolfyn/test_data/vector_data_imu01.log`
3. 3.05 MB `examples/data/dolfyn/test_data/RDI_withBT.log`

These add about 11 MB of log data. Is there a way we could reduce these log file sizes such as checking against a smaller dataset (and therefore a smaller associated log file), checking only a portion of the produced log, or mocking the data somehow?

File: mhkit/dolfyn/adv/motion.py
Comment (ssolson) [inline mhkit/dolfyn/adv/motion.py:112]: I think it should be "safe" instead of save

**Review by ssolson — COMMENTED** [review]
A couple of comments for consideration. Still working my way through these so this is just what I have so far.

**Comment by jmcvey3** [comment]
> James thank you for submitting the PR. To start the review I have a couple of questions about large log files:
> 
> 1. 1.16 MB `examples/data/dolfyn/test_data/AWAC_test01.log`
> 2. 6.74 MB `examples/data/dolfyn/test_data/vector_data_imu01.log`
> 3. 3.05 MB `examples/data/dolfyn/test_data/RDI_withBT.log`
> 
> These add about 11 MB of log data. Is there a way we could reduce these log file sizes such as checking against a smaller dataset (and therefore a smaller associated log file), checking only a portion of the produced log, or mocking the data somehow?

Good point, yes I can reduce the size of these.

File: mhkit/dolfyn/adp/clean.py
Comment (jmcvey3) [reply mhkit/dolfyn/adp/clean.py:174]: It looks like it puts everything after the first colon in italics (https://mhkit-software.github.io/MHKiT/mhkit-python/api.dolfyn.html#io)

File: mhkit/dolfyn/adv/clean.py
Comment (jmcvey3) [reply mhkit/dolfyn/adv/clean.py:13]: Good catch

File: mhkit/dolfyn/io/rdi_lib.py
Comment (ssolson) [inline mhkit/dolfyn/io/rdi_lib.py:59]: Is this something you plan to bring back in the future?

File: mhkit/dolfyn/io/rdi_lib.py
Comment (ssolson) [inline mhkit/dolfyn/io/rdi_lib.py:120]: Similar to above is a checksum() class going to be added in a future update?

File: mhkit/dolfyn/io/rdi.py
Comment (ssolson) [inline mhkit/dolfyn/io/rdi.py:817]: Looks like this was replaced by the definition above. Do we want to keep this?

**Review by ssolson — COMMENTED** [review]
@jmcvey3 a couple of minor comments. Great job on getting the docstring consistent. 

Is there anything I missed that you would like to talk through before we approve the PR?

File: mhkit/dolfyn/adv/motion.py
Comment (jmcvey3) [reply mhkit/dolfyn/adv/motion.py:36]: No, there's two filter frequencies here. One for acceleration and one for velocity. The velocity filter frequency defaults to 1/3 of the acceleration frequency (0.03), so it's default is 0.01.

File: mhkit/dolfyn/adv/turbulence.py
Comment (jmcvey3) [reply mhkit/dolfyn/adv/turbulence.py:224]: Yes, turns out the name is immediately overwritten once a dataArray is assigned a variable name.

File: mhkit/dolfyn/io/base.py
Comment (jmcvey3) [reply mhkit/dolfyn/io/base.py:28]: Thanks, yes it is

File: mhkit/dolfyn/io/rdi_lib.py
Comment (jmcvey3) [reply mhkit/dolfyn/io/rdi_lib.py:120]: No this is old. Looks like I forgot to drop it

File: mhkit/dolfyn/io/rdi.py
Comment (jmcvey3) [reply mhkit/dolfyn/io/rdi.py:817]: I'll remove it for now from the main branch until it's needed. I'm not entirely sure why it's in their documentation.

**Comment by jmcvey3** [comment]
@ssolson In case I hadn't mentioned it, I think this PR should now be covered. I'll need to update the docs, but I believe it's easiest to wait until this is merged into Develop?

**Comment by ssolson** [comment]
Thanks James. I'm going to merge into Develop. Please update the Docs as you mentioned.

**Review by ssolson — APPROVED** [review]

**Pull request merged**
